File: Leetcode/Amazon spring 23 high frequency/Optimal Partition of String/optimal.py
class Solution:
    def partitionString(self, s: str) -> int:
        j = 0 
        track = {}
        count = 0
        for i in range(len(s)) :
            if s[i] not in track or track[s[i]] < j :
                track[s[i]] = i
            else :
                count += 1
                j = i
                track[s[i]] = i
        return count + 1

File: Leetcode/Amazon spring 23 high frequency/Optimal Partition of String/optimal.java
class Solution {
    public int partitionString(String s) {
        //HashSet<Character> hs=new HashSet<>();//temp
        int freq[]=new int[26];
        int count=0;

        for(int i=0;i<s.length();i++)
        {
            char ch=s.charAt(i);
            if(freq[ch-'a']==1)
            {
                count++;
                Arrays.fill(freq,0);
            }
            freq[ch-'a']=1;
        }
        return count+1;
    }
}
